src/Acuminator/Acuminator.Analyzers/StaticAnalysis/EventHandlerModifier/EventHandlerProtectedModifierFix.cs

```diff
@@ -0,0 +1,193 @@
+			var isExplicitInterfaceImplementation = GetPropertyValue(diagnostic, "IsExplicitInterfaceImplementation");
+			var implementsInterface = GetPropertyValue(diagnostic, "ImplementsInterface");
+			var isContainingTypeSealed = GetPropertyValue(diagnostic, "IsContainingTypeSealed");
+
+			context.CancellationToken.ThrowIfCancellationRequested();
+
+			if (isOverride)
+			{
+				return;
+			}
+
+			if (isExplicitInterfaceImplementation)
+			{
+				var removeExplicitInterface = nameof(Resources.PX1077Fix_RemoveExplicitInterface).GetLocalized().ToString();
+				var codeFixAction = new RemoveExplicitInterfaceAction(removeExplicitInterface, context.Document, node);
+
+				context.RegisterCodeFix(codeFixAction, diagnostic);
+			}
+			else
+			{
+				if (!implementsInterface)
+				{
+					var accessibilityModifier = isContainingTypeSealed
+						? SyntaxKind.PublicKeyword
+						: SyntaxKind.ProtectedKeyword;
+
+					var addVirtualModifier = !isContainingTypeSealed;
+
+					var localizedText = SyntaxFactory.Token(accessibilityModifier).Text;
+					if (addVirtualModifier)
+					{
+						localizedText += " " + SyntaxFactory.Token(SyntaxKind.VirtualKeyword).Text;
+					}
+
+					var makeProtectedTitle = nameof(Resources.PX1077Fix).GetLocalized(localizedText).ToString();
+					var codeFixAction = new ChangeAccessibilityModifierAction(makeProtectedTitle, context.Document, node, accessibilityModifier, addVirtualModifier);
+
+					context.RegisterCodeFix(codeFixAction, diagnostic);
+				}
+			}
+		}
+
+		private static bool GetPropertyValue(Diagnostic diagnostic, string propertyName)
+		{
+			if (diagnostic.TryGetPropertyValue(propertyName, out string? strValue) && bool.TryParse(strValue, out bool value))
+			{
+				return value;
+			}
+
+			return false;
+		}
+
+		internal class ChangeAccessibilityModifierAction : CodeAction
+		{
+			private readonly string _title;
+			private readonly Document _document;
+			private readonly MethodDeclarationSyntax _method;
+			private readonly SyntaxKind _accessibilityModifier;
+			private readonly bool _addVirtual;
+
+			private static readonly List<SyntaxKind> SyntaxKinds =
+			[
+				SyntaxKind.PrivateKeyword,
+				SyntaxKind.PublicKeyword,
+				SyntaxKind.ProtectedKeyword,
+				SyntaxKind.InternalKeyword,
+				SyntaxKind.VirtualKeyword,
+				SyntaxKind.OverrideKeyword
+			];
+
+			public override string Title => _title;
+			public override string EquivalenceKey => _title;
+
+			public ChangeAccessibilityModifierAction(string title, Document document, MethodDeclarationSyntax method, SyntaxKind accessibilityModifier, bool addVirtual)
+			{
+				_title = title;
+				_document = document;
+				_method = method;
+				_accessibilityModifier = accessibilityModifier;
+				_addVirtual = addVirtual;
+			}
+
+			protected override async Task<Document> GetChangedDocumentAsync(CancellationToken cancellationToken)
+			{
+				var index = -1;
+				for (var i = 0; i < _method.Modifiers.Count; i++)
```

Comment:
This part will make a good separate `GetModifierIndex` method to follow SRP.

Reply:
Done.